### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by amrendrakind**
**Catalog of my things: Project requirements**
- Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
- All properties visible in the diagram should be defined and set up in the constructor method. Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.
- All methods visible in the diagram should be implemented:
  - add_item method (in all classes that have that method)
     - should take an instance of the Item class as an input.
     - should add the input item to the collection of items.
     - should add self as a property of the item object (by using the correct setter from the item object).
  - can_be_archived?() in the Item class
    - should return true if published_date is older than 10 years.
    - otherwise, it should return false.
  - move_to_archive() in the Item class
    - should reuse can_be_archived?() method.
    - should change the archived property to true if the result of the can_be_archived?() method is true.
    - should do nothing if the result of the can_be_archived?() method is false.
  - can_be_archived?() in the Book class
    - should override the method from the parent class.
    - should return true if parent's method returns true OR if cover_state equals to "bad".
    - otherwise, it should return false.
  - can_be_archived?() in the MusicAlbum class
    - should override the method from the parent class.
    - should return true if parent's method returns true AND if on_spotify equals true.
    - otherwise, it should return false.
  - can_be_archived?() in the Movie class
    - should override the method from the parent class.
    - should return true if parent's method returns true OR if silent equals true.
    - otherwise, it should return false.
  - can_be_archived?() in the Game class
    - should override the method from the parent class.
    - should return true if parent's method returns true AND if last_played_at is older than 2 years.
    - otherwise, it should return false.
- Add unit tests for all implemented methods.

**User interface**
- Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
   - Present the user with a list of options to perform.
   - Let users choose an option.
   - If needed, ask for parameters for the option.
   - Have a way to quit the app.
- The following options should be available:
   - List all books
   - List all music albums
   - List of games
   - List all genres
   - List all labels 
   - List all authors
   - Add a book
   - Add a music album
   - Add a game
- All data is preserved by saving collections in .json files.
**Screen Shot of Unit Tests**
Screenshot from 2022-07-08 02-23-02

**Video Link**
[Video Link for Project](https://drive.google.com/file/d/1BTrB2a6VURauMWX87LerCRGh_hJlQqGQ/view?usp=sharing)

### CODE REVIEW

File: classes/label.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def initialize(title, color) # rubocop:disable Lint/MissingSuper
+    @id = Random.rand(1..1000)
+    @title = title
+    @color = color
+    @items = []
+  end
+
+  def add_item(item)
+    item.add_label(self)
```
Comment (mikethreels) [inline classes/label.rb:15]: - [ ] this line should be like `item.label = self`

File: classes/label.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    @items << item unless @items.include?(item)
```
Comment (mikethreels) [inline classes/label.rb:16]: - [ ] in this line of the one to many relationship it is not needed to include the unless.

File: classes/music_album.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :on_spotify
+
+  def initialize(published_date, on_spotify)
+    super(published_date)
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    super and @on_spotify == true
+  end
```
Comment (mikethreels) [inline classes/music_album.rb:13]: - [ ] The `and` and `or` keywords are banned. It's just not worth it. Always use `&&` and `||` instead as mentioned in the [Ruby best practices](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md)

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] [OPTIONAL] when entering a wrong date format the application breaks, it would be nice if this wasn't the case.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by amrendrakind** [comment]
@mikethreels Thank you for your valuable feedback. We will implement your feedback.

**Review by EricMbouwe — APPROVED** [review]
Hi @amrendrakind, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**